tests/xtimer_usleep/main.c

```diff
@@ -24,35 +26,77 @@
 
 #include "xtimer.h"
 #include "timex.h"
+#include "stdlib.h"
 
 #define RUNS                (5U)
 #define SLEEP_TIMES_NUMOF   (sizeof(sleep_times) / sizeof(sleep_times[0]))
 
-static const uint32_t sleep_times[] = { 10000, 50000, 100000 };
+static const uint32_t sleep_times[] = { 10000, 50000, 1234, 5678, 121212, 98765, 75000};
 
+#define ERROR_US 500
+
+/*
+ * To use a pin to probe the sleep times enable the gpio module and define
+ * respective SLEEP_GPIO_PIN and set SLEEP_PIN to (1)
+ * */
+
+#define SLEEP_PIN (0)
+#if SLEEP_PIN
+#include "board.h"
+#include "periph/gpio.h"
+#define SLEEP_GPIO_PIN  GPIO_PIN(PORT_F, 7)
```

Comment:
This should be surrounded by an `#ifndef` and the pin number should be passed from the outside (Makefile or command line) if you want to use it in this way. Hardcoded GPIO might lead to failures e.g. while flashing on certain devices. The default pin might be `GPIO_UNDEF`,  but still I'm not convinced this is a generic way to perform a test like this.